setup.py

```diff
@@ -17,9 +19,53 @@ def get_dev_timestamp(version: ScmVersion) -> str:
     return version.format_with(f"{{tag}}.dev{int(time())}")
 
 
-setup(
-    use_scm_version={
-        "version_scheme": get_dev_timestamp,
-        "local_scheme": "dirty-tag",
-    },
-)
+def get_platform() -> str:
+    """Return the platform on which this code is run
+
+    Returns the platform name stripped of unnecessary information if the platform is known.
+    Otherwise, returns the full platform name.
+    """
+
+    known_platforms = {"windows", "linux", "darwin", "cygwin"}
+
+    target = platform.system().lower()
+    for known in known_platforms:
+        if target.startswith(known):
+            return known
+
+    return target
+
+
+def create_extension() -> setuptools.Extension:
+    """Create the ANTLR C++ extension to be passed to `setuptools.setup`"""
+
+    extra_compile_args = {
+        "windows": ["/DANTLR4CPP_STATIC", "/Zc:__cplusplus"],
+        "linux": ["-std=c++11"],
+        "darwin": ["-std=c++11"],
+        "cygwin": ["-std=c++11"],
+    }
+
+    sources = glob.glob("src/pytsql/grammar/cpp_src/**/*.cpp", recursive=True)
```

Comment:
Did you consciously decide against the `pathlib` module in favor of the `glob` module?

Reply:
Yes, using `glob` just resulted in less code.